### PR TITLE
▶Buttons pygame

### LINKED ISSUE #904: Buttons Using Pygame🆕
## Aim
- How to make buttons in python 
- Why to use them
- To be able to design and animate buttons in python .
- Make buttons use in any GUI based applications . ( It will be preety handy )
- To Redirect user to any other module using buttons .

## Details 
- Libraries to be used [Pygame]

### PR DESCRIPTION
**Pull request by Lakhankumawat**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/Lakhankumawat/Amazing-Python-Scripts/blob/Buttons-Pygame/Pygame-Buttons/README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [x] Games
- [ ] Networking
- [ ] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [x] Fun
- [ ] Others

Title: Pygame-Buttons

Folder: Pygame-Buttons

Requirements: requirements.txt

Script: main.py

Arguments: none

Contributor: Lakhan Kumawat

Description: How to make design and animate buttons in python.

### CODE REVIEW

**Comment by Lakhankumawat** [comment]
### 🙇Apologies beforehand
> Although my code works fine 
- As when you will go through code you'll see i haven't Added comments in between code wherever needed so it is difficult to understood
- Not added  more spacing between codes to look it bit more nicerer
- For any changes i will respond lately cause i have my exams going on but i'll do respond
- I don't know how to control the flow of functions in another py file which we are using as module from a py file thast's the reason if you'll see in my main file i import module only when a button is clicked and not at first , i know this is not a good practice to do it but seems like i tried and this was the only option i had .

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
For importing other file as module, to the following:
- Create an empty file called `__init__.py` in your project folder.
- Import your ExampleButtons on the top!

Explanation for this: `__init__.py` makes your directory act as a package. From package, you can import any module like we use external libraries. 

Also, The script didn't worked (error with color definition, possible solution: tuple definition of colors (255, 0, 0) for red and so on):

image

**Comment by Lakhankumawat** [comment]
okay i will try to fix them , infact just writing color name works in my spyder ide or maybe i m just using any module idk, i will make them in RGB format instead .
For importing modules i will learn it.

**Comment by Lakhankumawat** [comment]
Please have a look once again , although that module on i will solve later .

**Review by kaustubhgupta — COMMENTED** [review]
I am still not able to run your code:

image

**Comment by Lakhankumawat** [comment]
ot

Works fine in my case , i will look at it

**Review by kaustubhgupta — COMMENTED** [review]
I am still not able to run your code.

image

@santushtisharma10 @antrikshmisri can you confirm if this is only my error or you are also not able to run this?

**Comment by antrikshmisri** [comment]
> I am still not able to run your code.
> 
> image
> 
> @santushtisharma10 @antrikshmisri can you confirm if this is only my error or you are also not able to run this?

@kaustubhgupta The script is running fine
Screenshot (208)

**Comment by kaustubhgupta** [comment]
> > I am still not able to run your code.
> > image
> > @santushtisharma10 @antrikshmisri can you confirm if this is only my error or you are also not able to run this?
> 
> @kaustubhgupta The script is running fine
> Screenshot (208)

okay, mine issue then, apologies @Lakhankumawat! @antrikshmisri you can review this PR

**Comment by Lakhankumawat** [comment]
Make sure rest of all png buttons and others scripts attached are working fine too , they are working fine my side .

**Review by kaustubhgupta — APPROVED** [review]

**Comment by santushtisharma10** [comment]
image
image

@Lakhankumawat fix it

**Comment by Lakhankumawat** [comment]
In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable 
Also same goes for input cause i have to take input in single rect 

STK-20210415-WA0055-6

**Comment by Lakhankumawat** [comment]
Reviewers please review it or close it ....

**Review by santushtisharma10 — APPROVED** [review]

**Comment by antrikshmisri** [comment]
> In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable 
> Also same goes for input cause i have to take input in single rect 
> 
> 
> 
> 

You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable

**Comment by kaustubhgupta** [comment]
> > In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable
> > Also same goes for input cause i have to take input in single rect
> 
> You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable

@Lakhankumawat will this work?

**Comment by Lakhankumawat** [comment]
> > > In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable
> > > Also same goes for input cause i have to take input in single rect
> > 
> > 
> > You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable
> 
> @Lakhankumawat will this work?

Sir i tried but i could not get the way @antrikshmisri  sir told me to do . 
I tried my best and if you want to close pr then your choice but i tried . Ok

**Comment by kaustubhgupta** [comment]
> > > > In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable
> > > > Also same goes for input cause i have to take input in single rect
> > > 
> > > 
> > > You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable
> > 
> > 
> > @Lakhankumawat will this work?
> 
> Sir i tried but i could not get the way @antrikshmisri sir told me to do .
> I tried my best and if you want to close pr then your choice but i tried . Ok

No worries, I was just asking if the workaround worked or not. We will approve and merge this PR.

@antrikshmisri anything you would like to add?

**Comment by antrikshmisri** [comment]
@kaustubhgupta @Lakhankumawat No problem, we can merge it as is

**Review by antrikshmisri — APPROVED** [review]

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
@Lakhankumawat  Add the requirements.txt file and update the PR template with project metadata. Look at this PR for reference: https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/1002

**Comment by Lakhankumawat** [comment]
- Updated PR template with project metadata
- Added [Requirements.txt](https://github.com/Lakhankumawat/Amazing-Python-Scripts/blob/Buttons-Pygame/Pygame-Buttons/requirements.txt)

File: Pygame-Buttons/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+  | Requirement |  Command         | Documentation |
+--|-------------|------------------|---------------|
+1.| Pygame      |pip install pygame|  Pygame/PyPi  |
+--|-------------|------------------|---------------|
```
Comment (kaustubhgupta) [inline Pygame-Buttons/requirements.txt:4]: This not the way of defining the requirements file. See this requirement file: https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Amazon-Price-Alert/requirements.txt

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Pygame-Buttons/requirements.txt (hunk 1 above)
Comment (Lakhankumawat) [reply Pygame-Buttons/requirements.txt:4]: Sorry , i didn't knew that

**Comment by Lakhankumawat** [comment]
Changed requirements.txt as requested .

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**